src/serialize/jsonDeserializer.ts

```diff
@@ -0,0 +1,58 @@
+/*!
+This file is part of CycloneDX JavaScript Library.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+SPDX-License-Identifier: Apache-2.0
+Copyright (c) OWASP Foundation. All Rights Reserved.
+*/
+
+import type { Bom } from '../models'
+import { BaseDeserializer } from './baseDeserializer'
+import type { Factory as DenormalizerFactory } from './json/denormalize'
+import type { Normalized } from './json/types'
+import type { DenormalizerOptions, DesserializerOptions } from './types'
+
+/**
+ * Multi purpose Json deserializer.
+ */
+export class JsonDeserializer extends BaseDeserializer<Normalized.Bom> {
+  readonly #normalizerFactory: DenormalizerFactory
+
+  /**
+   * @throws {@link UnsupportedFormatError} if `normalizerFactory.spec` does not support {@link Format.JSON}.
+   */
+  constructor (normalizerFactory: JsonDeserializer['normalizerFactory']) {
+    super()
+    this.#normalizerFactory = normalizerFactory
+  }
+
+  get normalizerFactory (): DenormalizerFactory {
+    return this.#normalizerFactory
+  }
+
+  protected _denormalize (
+    data: Normalized.Bom,
+    options: DenormalizerOptions = {}
+  ): Bom {
+    return this.#normalizerFactory.makeForBom()
+      .denormalize(data, options)
+  }
+
+  protected _deserialize (
+    str: string,
+    opts: DesserializerOptions = {}
+  ): Normalized.Bom {
```

Comment:
:x: 
`JSON.parse()` can return anything, it is not guaranteed to return `Normalized.Bom`
The return type is under control of the users, because they control the input string.
Therefore, the algorithm is not robust.

Reply:
What do you propose?
Is there a JSON schema somewhere to validate against?

Reply:
Sure, schema file is included in the library. But there is no JSON schema validator included.

https://github.com/CycloneDX/cyclonedx-javascript-library/issues/86#issuecomment-1442152762

Reply:
closing, is obsolete